pkg/controllers/provisioningv2/rke2/machinedrain/machinedrain.go

```diff
@@ -35,35 +33,17 @@ type handler struct {
 func Register(ctx context.Context, clients *wrangler.Context) {
 	h := &handler{
 		ctx:          ctx,
+		machines:     clients.CAPI.Machine(),
 		machineCache: clients.CAPI.Machine().Cache(),
 		secrets:      clients.Core.Secret(),
 		secretCache:  clients.Core.Secret().Cache(),
 	}
 
-	// It would be a bad idea to have this handler run for every secret. Therefore, a new cache factory is created just for secrets with the machine plan type.
-	cacheFactory := cache.NewSharedCachedFactory(clients.ControllerFactory.SharedCacheFactory().SharedClientFactory(), &cache.SharedCacheFactoryOptions{
```

Comment:
I put this in previously because:
1. I thought I was being clever, and
2. I thought watching all secrets would be bad performance, but we are already doing that in several places.

So that is why I am removing this now.